tools/roslaunch/scripts/roslaunch-check

```diff
@@ -113,14 +113,14 @@ if __name__ == '__main__':
 
     if error_msg:
         print("FAILURE:\n%s"%error_msg, file=sys.stderr)
-        with open(test_file, 'w') as f:
+        with open(test_file, 'wb') as f:
```

Comment:
I believe this is the intended use given that the `junitxml` functions explicitly return encoded strings/bytes, see:

https://github.com/ros/ros/blob/9f2051dd809352fa11bab4671f47d95129125a2e/tools/rosunit/src/rosunit/junitxml.py#L495

And the xml output elsewhere has tests which confirm that it returns bytes, eg: https://github.com/ros/ros/pull/212